source/features/clear-pr-merge-commit-message.tsx

```diff
@@ -23,11 +25,15 @@ async function init(): Promise<void | false> {
 	}
 
 	// Preserve closing issues numbers when a PR is merged into a non-default branch since GitHub doesn't close them #4531
-	if (getBranches().base !== await getDefaultBranch()) {
-		for (const keyword of select.all('.comment-body .issue-keyword[aria-label^="This pull request closes"]')) {
-			const closingKeyword = keyword.textContent!.trim(); // Keep the keyword as-is (closes, fixes, etc.)
-			const issueLink = keyword.nextElementSibling as HTMLAnchorElement; // Account for issues not in the same repo
-			preservedContent.add(closingKeyword + ' ' + issueLink.href);
+	if (!await isPrAgainstDefaultBranch()) {
+		console.log(originalMessage, ...originalMessage.matchAll(/(fix(es|ed)?|close[sd]?|resolve[sd]?)([^\n]+)/gi));
+
+		// https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue
+		for (const [line] of originalMessage.matchAll(/(fix(es|ed)?|close[sd]?|resolve[sd]?)([^\n]+)/gi)) {
+			// Ensure it includes a reference or URL
+			if (/#\d+/.test(line) || line.includes('http')) {
```

Comment:
This restores my original suggestion, but looser https://github.com/refined-github/refined-github/issues/4531#issuecomment-986263580

All of this code exists to "clear the field, _except some data_". The DOM-based approach:

- adds data from other parts, which was not part of GitHub’s suggested message
- breaks more easily (as seen)

This regex will not _cleanly_ keep just the keywords, but that's fine, because that content was already in the field.